bld/CLMBuildNamelist.pm

```diff
@@ -4508,21 +4509,35 @@ sub setup_logic_fates {
     if (&value_is_true( $nl_flags->{'use_fates'})  ) {
         add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'fates_paramfile', 'phys'=>$nl_flags->{'phys'});
         my @list  = (  "fates_spitfire_mode", "use_fates_planthydro", "use_fates_ed_st3", "use_fates_ed_prescribed_phys",
-                       "use_fates_inventory_init","use_fates_fixed_biogeog","use_fates_nocomp","fates_seeddisp_cadence",
-                       "use_fates_logging","fates_parteh_mode", "use_fates_cohort_age_tracking","use_fates_tree_damage",
-		       "use_fates_luh","fates_history_dimlevel" );
+                       "use_fates_inventory_init","fates_seeddisp_cadence","fates_history_dimlevel",
+                       "fates_harvest_mode","fates_parteh_mode", "use_fates_cohort_age_tracking","use_fates_tree_damage" );
+
         foreach my $var ( @list ) {
- 	  add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, $var, 'use_fates'=>$nl_flags->{'use_fates'},
-                      'use_fates_sp'=>$nl_flags->{'use_fates_sp'} );
+           add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, $var, 'use_fates'=>$nl_flags->{'use_fates'},
+                       'use_fates_sp'=>$nl_flags->{'use_fates_sp'} );
         }
+
+        add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'use_fates_potentialveg', 'use_fates'=>$nl_flags->{'use_fates'});
+        add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'use_fates_lupft', 'use_fates'=>$nl_flags->{'use_fates'});
+        add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'use_fates_luh', 'use_fates'=>$nl_flags->{'use_fates'},
+                    'use_fates_lupft'=>$nl->get_value('use_fates_lupft'),
+                    'use_fates_potentialveg'=>$nl->get_value('use_fates_potentialveg'),
+                    'fates_harvest_mode'=>$nl->get_value('fates_harvest_mode') );
+        add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'use_fates_nocomp', 'use_fates'=>$nl_flags->{'use_fates'},
```

Comment:
@ekluzek is the way that I'm passing the value of `use_fates_lupft` here an ok way to make sure the nocomp default is appropriately set?  I know this way works as it passes unit tests, but it's different from the way `use_fates_sp` is passed in (i.e. by being added to the `nl_flags` list and referenced that way), so I wanted to double check.

Reply:
@glemieux for use_fates_sp it's put into nl_flags so that it can be used elsewhere as there is other logic and settings throughout the namelist that use it. The other reason for how it's handled is to make sure the the string match will be consistent. In FORTRAN .true., TRUE, .T., .t. are all the same, but interpreted as strings in the perl code they are different. The matches in the XML file is for ".true.", so putting it into nl_flags makes sure only that form is used.

Reply:
Copy that.  Thanks for clarifying the latter reason as well, I hadn't considered that.

Reply:
#2643

Reply:
Oh, I just realized that I think that the harvest_mode needs a removal of the leading and trailing quotes...

So something like...

``` perl
        add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, 'use_fates_luh', 'use_fates'=>$nl_flags->{'use_fates'},
                    'use_fates_lupft'=>$nl->get_value('use_fates_lupft'),
                    'use_fates_potentialveg'=>$nl->get_value('use_fates_potentialveg'),
                    'fates_harvest_mode'=>remove_leading_and_trailing_quotes($nl>get_value('fates_harvest_mode')) );
```

Otherwise I'm pretty sure it won't match as the actual string sent will have quotes around it and the XML needs an exact match. We should test the behavior to make sure it's doing the right thing.

Reply:
I confirmed that the current `use_fates_luh` does indeed fail if only `fates_harvest_mode = 'luhdata_area'` is set in the user_nl_clm file.  Thankfully it fails gracefully due to the added namelist build check in which the leading and training quotes are handled correctly.  I will update with the above correction.

Reply:
corrected per 9ee75610973c32376f6dc407c99a0bc8d55cad83